villain/ast.rkt

```diff
@@ -73,9 +91,15 @@
 (struct Let   (xs es e)       #:prefab)
 (struct Var   (x)             #:prefab)
 (struct App   (f es)          #:prefab)
+(struct LCall (e es)          #:prefab)
 (struct Apply (f e)           #:prefab)
 (struct Match (e0 cs)         #:prefab)
 (struct Vec   (es)            #:prefab)
+(struct Lam   (xs e)          #:prefab)
+(struct Lam-l (l xs e)        #:prefab)
```

Comment:
Does there need to be a labelled and unlabelled form?  Can't the parser insert labels so only the labelled form is needed?

Reply:
Yes, thanks!  I will change the code such that the parser does the transformation of the program and labelling of the lambdas.